src/nix/installables.cc

```diff
@@ -58,9 +58,32 @@ ref<EvalState> EvalCommand::getEvalState()
 {
     if (!evalState)
         evalState = std::make_shared<EvalState>(searchPath, getStore());
+    evalState->addRegistryOverrides(registryOverrides);
+    evalState->addLockFileOverrides(lockFileOverrides);
     return ref<EvalState>(evalState);
 }
 
+EvalCommand::EvalCommand()
+{
+    mkFlag()
+      .longName("override-flake")
+      .labels({"original-ref", "resolved-ref"})
+      .description("override a flake registry value")
+      .arity(2)
+      .handler([&](std::vector<std::string> ss) {
+          registryOverrides.insert_or_assign(FlakeRef(ss[0]), FlakeRef(ss[1]));
+      });
+
+    mkFlag()
+      .longName("override-lockfile")
```

Comment:
I think `--override-lock` is a better name, since we're overriding a specific lock, not an entire lockfile.